code/game/objects/items/binoculars.dm

```diff
@@ -161,25 +172,64 @@
 	to_chat(user, "<span class='notice'>INITIATING LASER TARGETING. Stand still.</span>")
 	if(!do_after(user, max(1.5 SECONDS, target_acquisition_delay - (2.5 SECONDS * user.skills.getRating("leadership"))), TRUE, TU, BUSY_ICON_GENERIC) || world.time < laser_cooldown || laser)
 		return
-	if(mode)
-		var/obj/effect/overlay/temp/laser_coordinate/LT = new (TU, laz_name, S)
-		coord = LT
-		to_chat(user, "<span class='notice'>SIMPLIFIED COORDINATES OF TARGET. LONGITUDE [coord.x]. LATITUDE [coord.y].</span>")
-		playsound(src, 'sound/effects/binoctarget.ogg', 35)
-		while(coord)
-			if(!do_after(user, 50, TRUE, coord, BUSY_ICON_GENERIC))
-				QDEL_NULL(coord)
-				break
-	else
-		to_chat(user, "<span class='notice'>TARGET ACQUIRED. LASER TARGETING IS ONLINE. DON'T MOVE.</span>")
-		var/obj/effect/overlay/temp/laser_target/LT = new (TU, laz_name, S)
-		laser = LT
-		playsound(src, 'sound/effects/binoctarget.ogg', 35)
-		while(laser)
-			if(!do_after(user, 50, TRUE, laser, BUSY_ICON_GENERIC))
-				QDEL_NULL(laser)
-				break
-
+	switch(mode)
+		if(0)
+			to_chat(user, "<span class='notice'>TARGET ACQUIRED. LASER TARGETING IS ONLINE. DON'T MOVE.</span>")
+			var/obj/effect/overlay/temp/laser_target/LT = new (TU, laz_name, S)
+			laser = LT
+			playsound(src, 'sound/effects/binoctarget.ogg', 35)
+			while(laser)
+				if(!do_after(user, 5 SECONDS, TRUE, laser, BUSY_ICON_GENERIC))
+					QDEL_NULL(laser)
+					break
+		if(1)
+			var/obj/effect/overlay/temp/laser_coordinate/LT = new (TU, laz_name, S)
+			coord = LT
+			to_chat(user, "<span class='notice'>SIMPLIFIED COORDINATES OF TARGET. LONGITUDE [coord.x]. LATITUDE [coord.y].</span>")
+			playsound(src, 'sound/effects/binoctarget.ogg', 35)
+			while(coord)
+				if(!do_after(user, 5 SECONDS, TRUE, coord, BUSY_ICON_GENERIC))
+					QDEL_NULL(coord)
+					break
+		if(2)
+			to_chat(user, "<span class='notice'>ACQUIRING TARGET. RAILGUN TRIANGULATING. DON'T MOVE.</span>")
+			if((GLOB.marine_main_ship?.rail_gun?.last_firing + 120 SECONDS) > world.time)
+				to_chat(usr, "[icon2html(src, usr)] <span class='warning'>The Rail Gun hasn't cooled down yet!</span>")
```

Comment:
Try to watch your usage of `usr` vs `user`.
You want `user` in all cases.

Reply:
Most cases, apparently the examine proc does usr.